src/include/reltoaltrep.hpp

```diff
@@ -17,13 +17,22 @@ struct AltrepRelationWrapper {
 
 	duckdb::unique_ptr<QueryResult> Materialize();
 
+	void MarkColumnAsTransformed();
+
 	const bool allow_materialization;
 	const size_t n_rows;
 	const size_t n_cells;
 
 	rel_extptr_t rel_eptr;
 	duckdb::shared_ptr<Relation> rel;
 	duckdb::unique_ptr<QueryResult> res;
+
+	R_xlen_t rowcount;
+	bool rowcount_retrieved;
```

Comment:
Is it worth to initialize with `-1` to avoid `rowcount_retrieved` ? I honestly don't know.